src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -121,15 +121,56 @@ bool DefaultOTARequestorDriver::ProviderLocationsEqual(const ProviderLocationTyp
     }
 }
 
-void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error) {}
+void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error)
+{
+    if (error == CHIP_NO_ERROR)
+    {
+        return;
+    }
+
+    switch (state)
+    {
+    case UpdateFailureState::kUnknown:
+        // Retry with next provider in the list
+        ScheduleRetry(false);
```

Comment:
Shouldn't we care if a retry was scheduled or not? Don't we want to start the periodic timer if we couldn't reschedule? Also, if we did successfully reschedule, shouldn't the caller of `HandleError` know this so they don't do something that could cause some other time to start, e.g. transition to idle state and cause periodic timer to be scheduled?

Reply:
Removed HandleError() and just let periodic timer kick in.